### ISSUE TITLE
Add a RabbitMQ option to disable FHC's read/write buffers

### ISSUE THREAD

**Issue by dumbbell** [rabbitmq/rabbitmq-server#226]
This will allow user to test how RabbitMQ behaves with their workload when especially the FHC read buffer is disabled. The results will be used to determine if we can get rid of it.

**Comment by dumbbell** [comment]
As shown in the commit message, here is how to disable read buffering in the FHC:
```erlang
%% In /etc/rabbitmq/rabbitmq.config
[   
  {rabbit, [
    {fhc_read_buffering, false}
  ]}  
].  
```

Buffering is still enabled by default.

**Comment by videlalvaro** [comment]
Can you compare how disabling the FHC Buffer affects this: https://github.com/rabbitmq/rabbitmq-server/issues/227

The case is publishers with 20k + msgs/sec in the 1kb size, without consumers.

**Comment by dumbbell** [comment]
@videlalvaro: I used the test you described in #227. I set a low memory watermark for trigger paging earlier and the storage is a tmpfs.

Disabling read and/or write buffering does not fix or change the problem.

**Comment by dumbbell** [comment]
After a suggestion from @michaelklishin, a log message tells if FHC buffering is turned on or not:
```
=INFO REPORT==== 24-Jul-2015::20:00:06 ===
FHC read buffering:  ON
FHC write buffering: ON
```

The message is logged during startup.

**Comment by michaelklishin** [comment]
Fixed by #229.
